src/safe-upgrade-tool/developer.md

```diff
@@ -0,0 +1,150 @@
+
+1. Open a terminal and navigate to the `graphql-schema-compatibility/` directory.
+1. Install project dependencies:
+
+   ```bash
+   npm install
+   ```
+
+1. Run unit tests or `jest`:
+
+   ```bash
+   npm run unit-test
+   ```
+
+The tests are inside `graphql-schema-compatibility/test/js/unit`.
+
+The string schemas for testing are inside `dev/graphql_schemas`.
+
+## Complexity score
+
+The **complexity score** is a figure that indicates how difficult an upgrade from the current version to the new one might be. Lower numbers indicate easier upgrades.
+
+{:.bs-callout-info}
+Complexity scores range between 0 and 100.
+
+This score is based on the results extracted from the analysis:
+
+*  Number of issues identified
+*  Severity of issues identified
+
+The SUT calculates this score according to the following formula:
+
+### Complexity score formula
+
+`Complexity Score = 2 * (# of errors) + 1 * (# of warnings)`
+
+## Packaging
+
+A package is essentially a directory containing something. It contains a package description which has a name and a version. The name and the version are used to identify the package.
+
+It is possible to create a `sut.zip` file of the SUT to distribute it with partners as a composer package.
+
+In order to upload it to be able to distribute it, you need first to create the package:
+
+1. To package the SUT execute the following command:
+
+   ```bash
+   bin/package VERSION`
+   ```
+
+### An example of a package
+
+`bin/package 0.0.1`
+
+1. This generates an `sut.zip` file that you can upload (or distribute with) to the [Magento Composer repository](https://repo.magento.com).
+
+{:.bs-callout-info}
+A repository is a package source. Composer will look in all your repositories to find the packages your SUT requires.
+
+If you need access to the [Magento Composer repository](https://repo.magento.com), submit a [Magento Support ticket]({{site.baseurl}}/cloud/trouble/trouble.html) to request permissions and an account.
+
+For more information about Composer, see the [Introduction to Magento Composer]({{site.baseurl}}/guides/v2.4/extension-dev-guide/intro/intro-composer.html) topic.
+
+## Distribution
+
+Once the SUT package is generated, you can upload it to the [Magento Composer repository](https://repo.magento.com).
+
+In order to distribute SUT:
+
+1. Go to the [Magento create repo](https://repo.magento.com/admin/packagist/web/ceeerelease/create) topic and create a new release with the following parameters:
+
+   *  Name: `magento/safe-upgrade-tool-VERSION`
+
+1. Go to the [Magento upload repo](https://repo.magento.com/admin/upload_m2_version) topic to upload the `sut.zip` with the following parameters:
+
+   *  Extension package: `sut.zip`
+   *  Release: `magento/safe-upgrade-tool-VERSION`
+   *  Edition: CE
+
+1. Finally, go to the [Magento version repo](https://repo.magento.com/admin/packagist/web/version/list) topic to check that the new version has been created.
```

Comment:
```suggestion
1. Finally, go to the [Magento version repo](https://repo.magento.com/admin/packagist/web/version/list) page to check that the new version has been created.
```

Reply:
We need to remove all the package section as highlighted by @paliarush

Reply:
will remove all packaging section then

Reply:
what about distribution @sgonzalezmangana?

Reply:
removed it, as per other comments